catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/FormulaEditorEditTextTest.java

```diff
@@ -0,0 +1,407 @@
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(longClick());
+		onFormulaEditor()
+				.checkShows(" ");
+		onFormulaEditor()
+				.performEnterFormula("1234");
+		setAbsoluteCursorPosition(2);
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(longClick());
+		onView(withId(R.id.formula_editor_edit_field))
+				.check(matches(withText("34 ")));
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testFunctionDeletion() {
+		onView(withId(R.id.formula_editor_keyboard_function))
+				.perform(click());
+		onView(withText(containsString(random)))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_function))
+				.perform(click());
+
+		onView(allOf(withText(containsString(random)), withEffectiveVisibility(ViewMatchers.Visibility.VISIBLE)))
+				.perform(click());
+
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onView(withId(R.id.formula_editor_keyboard_delete))
+				.perform(click());
+		onFormulaEditor()
+				.checkShows(" ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testNumberInsertion() {
+
+		onFormulaEditor()
+				.performEnterFormula(".1");
+		onFormulaEditor()
+				.checkShows("0" + decimalMark + "1 ");
+		setAbsoluteCursorPosition(0);
+		onFormulaEditor()
+				.performEnterFormula("5");
+		setAbsoluteCursorPosition(3);
+		onFormulaEditor()
+				.performEnterFormula("6");
+		onFormulaEditor()
+				.checkShows("50" + decimalMark + "61 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testGoBackToDiscardChanges() {
+
+		onFormulaEditor()
+				.performEnterFormula("99.9+");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		onView(withText(no))
+				.perform(click());
+		checkToastMessage(R.string.formula_editor_changes_discarded);
+
+		onBrickAtPosition(1)
+				.checkShowsText("0 ");
+	}
+
+	@Category({Cat.CatrobatLanguage.class, Level.Smoke.class})
+	@Test
+	public void testDiscardDialog() {
+		onFormulaEditor()
+				.performEnterFormula("99");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		pressBack();
+
+		onFormulaEditor()
+				.performEnterFormula(".9");
+		pressBack();
+		onView(withText(getCurrentActivity().getString(R.string.formula_editor_discard_changes_dialog_title)))
+				.check(matches(isDisplayed()));
+		onView(withText(yes))
+				.perform(click());
+		checkToastMessage(R.string.formula_editor_changes_saved);
+
+		onView(allOf(withId(R.id.brick_change_size_by_edit_text), withEffectiveVisibility(Visibility.VISIBLE)))
```

Comment:
please access bricks via the wrapper methods. (eg. onBrickAtPosition... when in script view).
due to the pressBack() switching around its really hard to follow... 
the onBrickAtPosition would give the additional info that now were back in the scriptFragment